src/main/java/no/ecc/vectortile/VectorTileEncoder.java

```diff
@@ -169,11 +174,23 @@ public void addFeature(String layerName, Map<String, ?> attributes, Geometry geo
 
         layer.features.add(feature);
     }
-    
+
+    /**
+     * A short circuit clip to the tile extent (tile boundary + buffer) for points to improve performance.
+     * @see https://github.com/ElectronicChartCentre/java-vector-tile/issues/13
+     */
+    protected Geometry clipToExtent(Point geom) {
```

Comment:
Yes.

If you are in agreement, perhaps you could cherry pick from this PR to do a short circuited check if addFeature should continue for points?  I'm happy to do another PR, but it is really just 4 lines of code.

Thank you for paying attention to feedback - it's greatly appreciated.